### DISCUSSION TITLE
IP Routing in S8 Inbound Roaming

### DISCUSSION THREAD

**Discussion by jpad-freedomfi** [magma/magma#6765] [Q&A] [answered]
I had call today with an IPX/GRX provider today regarding using Magma for S8/s6A inbound roaming. The IPX vendor was asking about IP routing when GTP-C and GTP-U are coming from different IPs. Moreover how it works multiple GTP-U endpoints are signaled over the same single GTP-C endpoint.

The IPX vendor asked to see a call flow diagram and pcap of a session setup and data transfer. I see in https://github.com/magma/magma/blob/master/docs/readmes/proposals/p008_inbound_roaming_with_subscriberdb.md (and the detailed design doc linked in there) that the call flow covers GTP-C portion, but doesn't show GTP-U data. 

Questions:

1. How is routing envisioned for finding the right GTP-U endpoints for a given Create Session Request when multiple AGWs are distributed across sites?
2. Is there a pcap file that can be shared from testing this feature that shows GTP-C + GTP-U traffic? If we have that I can make the call flow diagram. 
3. Can the plant uml file be shared for the call flow diagram in the design google doc?

**Comment by jpad-freedomfi** [comment]
trying to tag people I think might be able to help. @ssanadhya and @uri200

**Comment by uri200** [comment, marked as answer]
Currently there is no plans of load balancing at Magma. Magma contacts one single PGW on the control plane (configured on the FEG). That PGW is responsible to provide Magma with the FTeid of the user plane, which includes the GTP U IP of the PGW. So the balancing is done by the home network.

GTP C is the one that sets up the connection. GTP U just sends the packets to the address received on the Create Session Response. That is why there is no call flow.

Note that PGW and Magma need to be able to reach each other through the user plane. That is why VPN is needed on GTP U.

Attached you can find a pcap with GTP-C and GTP-U
[inbound_roaming_01_210414_061714-1.pcapng.zip](https://github.com/magma/magma/files/6429361/inbound_roaming_01_210414_061714-1.pcapng.zip)




![image](https://user-images.githubusercontent.com/16157139/117189699-5868e800-ad93-11eb-9a76-ed5ad35cd830.png)


https://docs.magmacore.org/docs/howtos/inbound_roaming

**Reply by jpad-freedomfi** [reply]
Awesome, that is all very very helpful. Thanks @uri200
